seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisOutputFormat.java

```diff
@@ -0,0 +1,230 @@
+        this.dorisStreamLoad = dorisStreamLoad;
+        parseDelimiter();
+        this.fieldNames = fieldNames;
+        this.batchSize = batchSize;
+        this.batchIntervalMs = batchIntervalMs;
+        this.maxRetries = maxRetries;
+        this.jsonFormat = FORMAT_JSON_VALUE.equals(dorisStreamLoad.getStreamLoadProp().getProperty(FORMAT_KEY));
+    }
+
+    private void parseDelimiter() {
+        Properties streamLoadProp = dorisStreamLoad.getStreamLoadProp();
+        boolean ifEscape = Boolean.parseBoolean(streamLoadProp.getProperty(ESCAPE_DELIMITERS_KEY, ESCAPE_DELIMITERS_DEFAULT));
+        if (ifEscape) {
+            this.fieldDelimiter = escapeString(streamLoadProp.getProperty(FIELD_DELIMITER_KEY,
+                    FIELD_DELIMITER_DEFAULT));
+            this.lineDelimiter = escapeString(streamLoadProp.getProperty(LINE_DELIMITER_KEY,
+                    LINE_DELIMITER_DEFAULT));
+
+            if (streamLoadProp.contains(ESCAPE_DELIMITERS_KEY)) {
+                streamLoadProp.remove(ESCAPE_DELIMITERS_KEY);
+            }
+        } else {
+            this.fieldDelimiter = streamLoadProp.getProperty(FIELD_DELIMITER_KEY,
+                    FIELD_DELIMITER_DEFAULT);
+            this.lineDelimiter = streamLoadProp.getProperty(LINE_DELIMITER_KEY,
+                    LINE_DELIMITER_DEFAULT);
+        }
+    }
+
+    private String escapeString(String s) {
+        Matcher m = DELIMITER_PATTERN.matcher(s);
+        StringBuffer buf = new StringBuffer();
+        while (m.find()) {
+            m.appendReplacement(buf, String.format("%s", (char) Integer.parseInt(m.group(1))));
+        }
+        m.appendTail(buf);
+        return buf.toString();
+    }
+
+    @Override
+    public void configure(Configuration configuration) {
+    }
+
+    @Override
+    public void open(int taskNumber, int numTasks) {
+        if (batchIntervalMs != 0 && batchIntervalMs != 1) {
+            this.scheduler = new ScheduledThreadPoolExecutor(1, new ExecutorThreadFactory("doris-streamload-outputformat"));
+            this.scheduledFuture = this.scheduler.scheduleWithFixedDelay(() -> {
+                synchronized (DorisOutputFormat.this) {
+                    if (!closed) {
+                        try {
+                            flush();
+                        } catch (Exception e) {
+                            flushException = e;
+                        }
+                    }
+                }
+            }, batchIntervalMs, batchIntervalMs, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void checkFlushException() {
+        if (flushException != null) {
+            throw new RuntimeException("Writing records to streamload failed.", flushException);
+        }
+    }
+
+    @Override
+    public synchronized void writeRecord(T row) throws IOException {
+        checkFlushException();
+        addBatch(row);
+        if (batchSize > 0 && batch.size() >= batchSize) {
```

Comment:
I think we should make sure that `batchSize` is larger than 0 at begining.

Reply:
Good idea, i will modify